### ISSUE TITLE
Avoid GCC diagnostic ignored

### ISSUE THREAD

**Issue by jlguardi** [zeromq/libzmq#4374]
*Please use this template for reporting suspected bugs or requests for help.*

# Issue description

Following from #4206, I think this is not a false positive but a possible situation.

Probably, it could be better to perform something like the [patch](https://github.com/jlguardi/libzmq/commit/1520bd7a26487e868aaab9a5174780090086a7e8) attached

# Environment

* libzmq version (commit hash if unreleased): 572eb008f197e9d047b14a4e4cdfca41eb7c5ffd
* OS: Ubuntu 22.04

# Minimal test code / Steps to reproduce the issue

1.  ./configure --with-pgm --with-libsodium --with-norm=yes --with-libgssapi_krb5=yes
2. make

No warnings are generated with both original and patched code.

# What's the actual result? (include assertion message & call stack if applicable)
With current code, probably segfault could be produced if no memory is available.

**Comment by alajovic** [comment]
I don't think that the suggested problem actually exists. `initiate_plaintext` is a `std::vector`, and if no memory is available, its creation will either trigger an `alloc_assert` or throw a `std::bad_alloc` (both in `secure_allocator_t`, but depending on which implementation gets used). In such a case, the code block in question will not be executed at all.
